File: src/Dictionaries/Recommendations.Dictionaries.Shared/DTO/ProductDto.cs
namespace Recommendations.Dictionaries.Shared.DTO;

public sealed record ProductDto(
    Guid Id,
    string? ExternalId,
    string ProductDisplayName,
    string BrandName,
    Guid SubCategoryId,
    string SubCategoryName,
    Guid ArticleTypeId,
    string ArticleTypeName,
    Guid BaseColourId,
    string BaseColourName,
    decimal Price,
    decimal? OriginalPrice,
    decimal Rating,
    int Reviews,
    bool IsBestseller,
    bool IsNew,
    bool IsTrending,
    bool IsOnSale,
    bool ProfitBoost,
    ProductDetailsDto? Details,
    IReadOnlyCollection<ProductImageDto> Images); 

File: fe/src/types/product/ProductDto.ts
export interface ProductDto {
  id: string;
  productDisplayName: string;
  brandName: string;
  subCategoryId: string;
  subCategoryName: string;
  articleTypeId: string;
  articleTypeName: string;
  baseColourId: string;
  baseColourName: string;
  price: number;
  originalPrice?: number;
  rating: number;
  reviews: number;
  isBestseller: boolean;
  isNew: boolean;
  details?: ProductDetailsDto;
  images: ProductImageDto[];
}

export interface ProductDetailsDto {
  gender: string;
  season: string;
  usage: string;
  year?: string;
  description?: string;
  sleeveLength?: string;
  fit?: string;
  fabric?: string;
  collar?: string;
  bodyOrGarmentSize?: string;
  pattern?: string;
  ageGroup?: string;
}

export interface ProductImageDto {
  imageUrl: string;
  imageType: string;
  resolution?: string;
  isPrimary: boolean;
}

export interface ProductFullDto {
  product: ProductDto;
  details?: ProductDetailsDto;
  images: ProductImageDto[];
}

export interface SubCategoryDto {
  id: string;
  name: string;
  masterCategoryId: string;
  articleTypes: ArticleTypeDto[];
}

export interface MasterCategoryDto {
  id: string;
  name: string;
  subCategories: SubCategoryDto[];
}

export interface ArticleTypeDto {
  id: string;
  name: string;
  subCategoryId: string;
}

export interface BaseColourDto {
  id: string;
  name: string;
}
